### PR TITLE
[RHCLOUD-19727] Make sources support managed Kafka

### PR DESCRIPTION
**Pull request by MikelAlejoBR**
This PR adds support for managed Kafka. Basically what we do is:

* Read the Sasl authentication settings from Clowder.
* Apply those authentication settings to the kafka-go reader/writer.

When calling `.Consumer()` and `.Producer()` we can receive an error now due to the above changes, so that's why I created the handy `.CloseConsumer()` and `.CloseProducer()` functions. I'm welcome to refactoring this if you can think of any other way of doing this.

## Links

[[RHCLOUD-19727]](https://issues.redhat.com/browse/RHCLOUD-19727)

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -83,8 +89,22 @@ func Get() *SourcesApiConfig {
...
+
+		if kafkaBroker.Authtype != nil {
+			options.SetDefault("KafkaSaslEnabled", true)
+
+			caPath, err := cfg.KafkaCa(kafkaBroker)
+			if err != nil {
+				log.Fatalf(`Could not write the Kafka broker's CA to a temporary directory: %s`, err)
+			}
+
+			options.SetDefault("KafkaSaslCaPath", caPath)
```
Comment (lpichler) [inline config/config.go:102]: can we test those cars for emptiness/nil ? + log them in case of when there are empty/nil - it could help with potential debugging.

File: service/availability_check.go [hunk 2]
```diff
@@ -174,9 +174,9 @@ func publishSatelliteMessage(mgr *kafka.Manager, source *m.Source, endpoint *m.E
 		l.Log.Warnf("Failed to produce kafka message for Source %v, error: %v", source.ID, err)
 	}
 
-	err = mgr.Producer().Close()
+	err = mgr.CloseProducer()
 	if err != nil {
-		l.Log.Warnf("Failed to close kafka producer: %v", err)
+		l.Log.Errorf(`unable to close the Kafka producer when publishing a Satellite availability status check message: %s`, err)
```
Comment (lpichler) [inline service/availability_check.go:179]: is this related just to Satellite ?

**Comment by lpichler** [comment]
Looks to good to me I had just few comments - very straightforward and clear implementation 👏

File: service/availability_check.go (hunk 2 above)
Comment (MikelAlejoBR) [reply service/availability_check.go:179]: In this case the log message is in `service/availability_check#publishSatelliteMessaage` function, so that's why I thought that I could specify the error a little bit more.

File: config/config.go (hunk 1 above)
Comment (MikelAlejoBR) [reply config/config.go:102]: Sure, I've added those nil checks and logged the values except for the sensitive credentials. I was doubting if I should simply `Fatal` if any of the `mechanism`, `username` or `password` values are empty, since without those we wouldn't be able to connect to Kafka... Everything would work, but not Kafka.

**Review by lpichler — APPROVED** [review]
